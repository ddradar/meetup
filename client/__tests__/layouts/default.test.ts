import {
  createLocalVue,
  mount,
  RouterLinkStub,
  shallowMount,
  Wrapper,
} from '@vue/test-utils'
import Buefy from 'buefy'

import DefaultLayout from '~/layouts/default.vue'

const localVue = createLocalVue()
localVue.use(Buefy)

describe('/layouts/default.vue', () => {
  const stubs = { NuxtLink: RouterLinkStub, Nuxt: true }

  describe('snapshot test', () => {
    test('renders Login button if { isLoggedIn: false }', () => {
      // Arrange - Act
      const mocks = { $accessor: { isLoggedIn: false } }
      const wrapper = mount(DefaultLayout, { localVue, stubs, mocks })

      // Assert
      expect(wrapper.element).toMatchSnapshot()
    })
    test('renders Logout button if { isLoggedIn: true }', () => {
      // Arrange - Act
      const mocks = { $accessor: { isLoggedIn: true } }
      const wrapper = mount(DefaultLayout, { localVue, stubs, mocks })

      // Assert
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('methods', () => {
    const mocks = {
      $accessor: { isLoggedIn: false, login: jest.fn(), logout: jest.fn() },
    }
    let wrapper: Wrapper<DefaultLayout>
    beforeEach(() => {
      mocks.$accessor.login.mockClear()
      mocks.$accessor.logout.mockClear()
      wrapper = shallowMount(DefaultLayout, { localVue, stubs, mocks })
    })

    describe('login()', () => {
      test('calls $accessor.login()', async () => {
        // Arrange - Act
        // @ts-ignore
        await wrapper.vm.login()

        expect(mocks.$accessor.login).toBeCalledTimes(1)
        expect(mocks.$accessor.logout).not.toBeCalled()
      })
    })
    describe('logout()', () => {
      test('calls $accessor.logout()', async () => {
        // Arrange - Act
        // @ts-ignore
        await wrapper.vm.logout()

        expect(mocks.$accessor.logout).toBeCalledTimes(1)
        expect(mocks.$accessor.login).not.toBeCalled()
      })
    })
  })
})
