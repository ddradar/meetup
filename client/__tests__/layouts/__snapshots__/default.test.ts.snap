// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/layouts/default.vue snapshot test renders Login button if { isLoggedIn: false } 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar is-primary"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        class="navbar-item"
      >
        <img
          alt="Buefy"
          height="28"
          src="~assets/buefy.png"
        />
      </a>
      <a
        class="navbar-item"
      >
        <span
          class="icon"
        >
          <i
            class="mdi mdi-home mdi-24px"
          />
        </span>
         Home
      
      </a>
      <a
        class="navbar-item"
      >
        <span
          class="icon"
        >
          <i
            class="mdi mdi-lightbulb mdi-24px"
          />
        </span>
         Inspire
      
      </a>
    </div>
    <div
      class="navbar-menu"
    >
      <div
        class="navbar-start"
      />
      <div
        class="navbar-end"
      >
        <div
          class="navbar-item"
        >
          <div
            class="buttons"
          >
            <button
              class="button is-info"
              type="button"
            >
              <span
                class="icon is-small"
              >
                <i
                  class="mdi mdi-twitter"
                />
              </span>
              <span>
                ログイン
              </span>
              <!---->
            </button>
          </div>
        </div>
      </div>
    </div>
  </nav>
   
  <section
    class="main-content"
  >
    <nuxt-stub />
  </section>
   
  <footer
    class="footer"
  >
    <div
      class="content has-text-centered"
    >
      
      Powered by 
      <a
        href="https://www.ddradar.app/"
      >
        DDRadar
      </a>
    </div>
  </footer>
</div>
`;

exports[`/layouts/default.vue snapshot test renders Logout button if { isLoggedIn: true } 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar is-primary"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        class="navbar-item"
      >
        <img
          alt="Buefy"
          height="28"
          src="~assets/buefy.png"
        />
      </a>
      <a
        class="navbar-item"
      >
        <span
          class="icon"
        >
          <i
            class="mdi mdi-home mdi-24px"
          />
        </span>
         Home
      
      </a>
      <a
        class="navbar-item"
      >
        <span
          class="icon"
        >
          <i
            class="mdi mdi-lightbulb mdi-24px"
          />
        </span>
         Inspire
      
      </a>
    </div>
    <div
      class="navbar-menu"
    >
      <div
        class="navbar-start"
      />
      <div
        class="navbar-end"
      >
        <div
          class="navbar-item"
        >
          <div
            class="buttons"
          >
            <button
              class="button"
              type="button"
            >
              <!---->
              <span>
                ログアウト
              </span>
              <!---->
            </button>
          </div>
        </div>
      </div>
    </div>
  </nav>
   
  <section
    class="main-content"
  >
    <nuxt-stub />
  </section>
   
  <footer
    class="footer"
  >
    <div
      class="content has-text-centered"
    >
      
      Powered by 
      <a
        href="https://www.ddradar.app/"
      >
        DDRadar
      </a>
    </div>
  </footer>
</div>
`;
